### ISSUE TITLE
FunctionFactory fails with string parameter

### ISSUE THREAD

**Issue by matiassilioni** [seesharper/LightInject#515]
Updating to version 5.5 to 6.x i found this problem. 

Actually i have a function factory receiving a string to build an Azure Table access, so i pass the table name. Updating to 6.x this behavior is not working any more. Just add this test to verify.

```
public class FooWithOneStringParameter : IFoo
    {
        public string Arg1 { get; private set; }

        public FooWithOneStringParameter(string arg1)
        {
            Arg1 = arg1;
        }
    }

[Fact]
        public void Invoke_OneStringParameter_FactoryCreatesInstanceAndPassesValuesToConstructor()
        {
            var container = CreateContainer();
            container.Register<string, FooWithOneStringParameter>((factory, i) => new FooWithOneStringParameter(i));
            var fooFactory = container.GetInstance<Func<string, FooWithOneStringParameter>>();
            var instance = (FooWithOneStringParameter)fooFactory("1");
            Assert.Equal("1", instance.Arg1);
        }
```

**Comment by seesharper** [comment]
@matiassilioni In LightInject 6.0 received a major overhaul of the scoping mechanism and these factory methods taking runtime arguments were left out. 
They are most likely to be deprecated at some point as they simply cause nothing but trouble. The problem is that we start to mix runtime data with services. The error is caused by an ambiguity between a named service and a service taking a string runtime argument.

You can get around this problem by passing something else than string. 

Something like 

```
var container = new ServiceContainer();
container.Register<TableName, FooWithTableNameParameter>((factory, i) => new FooWithTableNameParameter(i));
var fooFactory = container.GetInstance<Func<TableName, FooWithTableNameParameter>>();
var instance = (FooWithTableNameParameter)fooFactory("sdfsdf");

public class FooWithTableNameParameter
{
    public string Arg1 { get; private set; }

    public FooWithTableNameParameter(string arg1)
    {
        Arg1 = arg1;
    }
}

public class TableName
{
    public TableName(string name)
    {
        Name = name;
    }

    public string Name { get; }

    public static implicit operator TableName(string s)
    {
        return new TableName(s);
    }

    public static implicit operator string(TableName s)
    {
        return s.Name;
    }
}
```

Note that the explicit operators allows us to still pass a string. 

That being said this is not a recommended approach. 

In your case it would probably be better to have a factory taking a table name 

I don't know much about Azure tables but maybe something like 

```
public interface ITableFactory 
{
    AzureTable GetTable(string tableName);
}
```

In other words, runtime data should be passed into methods and not as constructor arguments.

**Comment by matiassilioni** [comment]
Great! Thanks for your support, I'll take ITableFactory solution
